pkg/proxy/iptables/proxier.go

```diff
@@ -960,32 +960,55 @@ func (proxier *Proxier) syncProxyRules() {
 		if svcInfo.NodePort != 0 {
 			// Hold the local port open so no other process can open it
 			// (because the socket might open but it would never work).
-			lp := utilproxy.LocalPort{
-				Description: "nodePort for " + svcNameString,
-				IP:          "",
-				Port:        svcInfo.NodePort,
-				Protocol:    protocol,
+			addresses, err := utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
+			if err != nil {
+				glog.Errorf("Failed to get node ip address matching nodeport cidr")
```

Comment:
Change error to read something like: glog.Errorf("Failed to get nodeport ip addresses: %v", err). Basically, include the error from the GetNodeAddresses call in some way. Same comment for ipvs code

Reply:
Done.